legacy/src/main/java/org/opensearch/sql/legacy/executor/format/SelectResultSet.java

```diff
@@ -180,8 +185,12 @@ private void loadFromEsState(Query query) {
     Map<String, FieldMappingMetadata> typeMappings = mappings.get(indexName);
 
     this.indexName = this.indexName == null ? indexName : (this.indexName + "|" + indexName);
-    this.columns.addAll(
-        renameColumnWithTableAlias(query, populateColumns(query, fieldNames, typeMappings)));
+    if (isPartOfUnion(query) && !this.columns.isEmpty()) {
```

Comment:
What's the difference between `isUnionQuery` and `isPartOfUnion`? Do we need the new `partOfUnion` field?

Reply:
`isUnionQuery` checks if the query type is `Union`, just similar to `isJoinQuery` does, checking if the type is `JoinSelect`.
The children type of Join is `TableOnJoinSelect`, a special type rather than a normal `Select`. But The children type of Union is a normal `Select`, unless we set a new type for Union's children, we can't identify a `Select` is a normal one or a child of Union in `loadFromEsState`, that's the reason I add `partOfUnion` in Select. Or add a new class `class UnionSubSelect extends Select`. Which one would be better?  
https://github.com/opensearch-project/sql/pull/2757/files#diff-0b55e630b732c43e3c8ea8038eb068f21a3c46bcd1744c98be4cc94bad799b84R50